physcraper/__init__.py

```diff
@@ -1809,186 +1810,168 @@ def run_blast_wrapper(self):  # TODO Should this be happening elsewhere?
+                    
+                    # if gb acc was already read in before stop the for loop
+                    if gb_acc in query_dict or gb_acc in self.data.gb_dict:
+                        # debug("set to true")
+                        stop_while = True
+                        break
+                    sscinames = self.ids.ncbi_parser.get_name_from_id(staxids)
+                    # # likely not a speec up to just the single line above
+                    # # sometimes if multiple seqs are merged,
+                    # # we lack information about which taxon is which gb_acc...
+                    # # test it here:
+                    # # if we have same number of gb_acc and taxon id go ahead as usual
+                    # # debug("get name from id")
+                    # if len(sallseqid_l) == len(staxids_l):
+                    #     sscinames = sscinames_l[i]
+                    # # only one taxon id present, all are from same taxon
+                    # elif len(staxids_l) == 1:
+                    #     sscinames = sscinames_l[0]
+                    #     qtaxid = staxids_l[0]
+                    # # if not the first item and id different from before: get name
+                    # elif i != 0 and id_before != id_now:
+                    #     sscinames = self.ids.ncbi_parser.get_name_from_id(qtaxid)
+                    # elif i == 0:  # for first item in redundant data, always get info
+                    #     sscinames = self.ids.ncbi_parser.get_name_from_id(qtaxid)
+                    # else:  # if id_before and id_now were the same, we do not need to add same seq again
+                    #     continue
+                    try:
+                        # assert str(staxids) in staxids_l, (str(staxids), staxids_l)
+                        # next vars are used to stop loop if all taxids were found
+                        found_taxids.add(staxids)
+                        found_spn.add(sscinames)
+                        if gb_acc not in query_dict and gb_acc not in self.newseqs_acc:
+                            query_dict[gb_acc] = \
+                                {'^ncbi:gi': gi_id, 'accession': gb_acc, 'staxids': staxids,
+                                 'sscinames': sscinames, 'pident': pident, 'evalue': evalue,
+                                 'bitscore': bitscore, 'sseq': sseq, 'title': stitle}
+                    except AssertionError:
+                        sys.stderr.write("Taxon Id {} was not initally found in the blast search - corresponding Genbank accession number is: {}. Sequence will not be added. Check with ncbi. \n".format(staxids, gb_acc))
+         
+            # same loop as above, only that it does more blastdbcmd's
+            # this is used as sometimes the if above does not yield in stop_while == True,
+            # through different taxa names
+            elif count >= 1 and stop_while is False:
```

Comment:
I see the comment above - but why would the original loop not get all the taxon names? Is it posible to moidify it, so that we don't need to come back to it here?